addons/im_livechat/static/src/embed/core/thread_service_patch.js

```diff
@@ -63,20 +63,10 @@ patch(ThreadService.prototype, {
         }
     },
 
-    avatarUrl(author, thread) {
-        if (thread?.type !== "livechat") {
-            return super.avatarUrl(...arguments);
-        }
-        const isFromOperator =
-            author && author.id !== this.livechatService.options.current_partner_id;
-        if (isFromOperator) {
-            return `${session.origin}/im_livechat/operator/${
-                author?.id ?? thread.operator.id
-            }/avatar`;
-        } else if (author) {
-            return `${session.origin}/web/image/res.partner/${author.id}/avatar_128`;
-        } else {
-            return `${session.origin}/mail/static/src/img/smiley/avatar.jpg`;
+    avatarUrl(persona, thread) {
+        if (thread.type === "livechat" && persona.eq(thread.operator)) {
+            return url(`/im_livechat/operator/${persona.id}/avatar`);
```

Comment:
Do we still need the operator specific route? Now there is a guest, the channel route should always work with the token.

Reply:
The operator avatar should be available even before the channel (and the guest) are created. So I think this route should be kept.